### ISSUE TITLE
ACT does not require that the language of text is indicated

### ISSUE THREAD

**Issue by bert-github** [w3c/wcag-act#573]
(This is part of the review by the Internationalization WG.)

4.2. Rule Description
https://www.w3.org/TR/2024/WD-act-rules-format-1.1-20240618/#rule-description

> a description that is in plain language

The description may be displayed or printed and to do that correctly, you need to know the language and the writing direction of the text. Which means that one requirement on the format used for storing rules should be that it must have a way to indicate the language and direction of text. (HTML is an example of a format that can do so.) And a requirement for the author of the description should be to make sure the language and direction of all text in the description are indeed specified.

Referring to [‘Strings on the Web: Language and Direction Metadata’](https://www.w3.org/TR/string-meta/), maybe in a note, may be useful. It explains why language and direction are needed and describes best practices.

**Comment by WilcoFiers** [comment]
@bert-github I believe this to be covered by the requirement that the rule be accessible. All standards for accessibility include a requirement that the language is programmatically determined. I think this is sufficient.

**Comment by daniel-montalvo** [comment]
Hi @bert-github 
Has the group had a chance to review @WilcoFiers  comments above? Would this explanation be sufficient to mark this as resolved?

**Comment by bert-github** [comment]
Yes, there is text in WCAG that says that it must be possible to determine the human language of text. But WCAG doesn't mention writing direction, which is also needed for correct display of text.

It is the I18N WG's opinion that it is better to have an explicit note somewhere in the spec itself. A note to remind people that, whenever the spec requires text in plain language, it assumes that text has metadata specifying its human language and writing direction. (How that is expressed in the document depends on the specific format.) That is because the text is not necessarily in English and the language and direction may be needed to display the text correctly.

It may be useful to include a link to [Use cases for bidi and language metadata on the Web](https://www.w3.org/International/articles/lang-bidi-use-cases/index.en.html), for examples of why language and direction are needed.

**Comment by daniel-montalvo** [comment]
Hi @aphillips and all.

See #585 for the proposed changes. If you agree, we'll merge them into the Editors Draft.

**Comment by aphillips** [comment]
The I18N WG reviewed the text in #585 in our teleconference of 2025-04-03, which say in part:

> Note: Regardless of the data format in which a rule is written, it is assumed that the rule has metadata specifying its language and writing direction. Most well-known markup languages have built-in provisions to satisfy this requirement.

We don't think this is quite correct? I'd suggest:

> Regardless of the data format in which a rule is written, the description needs to include metadata specifying its language and base paragraph direction. Most markup languages or document formats have built-in provisions for indicating these.

* This is _not_ a note, although I have avoided 2119 language.
* "writing direction" is imprecise. "base paragraph direction" or "string direction" are probably better terms
* You might also wish to include a link to our [string-meta](https://www.w3.org/TR/string-meta) (not shown)

**Comment by daniel-montalvo** [comment]
Hi @aphillips 
The group discussed this in our [10 April meeting](https://www.w3.org/2025/04/10-wcag-act-minutes.html) and think #592 addresses your concerns.
